lib/signaling/v2/twilioconnectiontransport.js

```diff
@@ -324,6 +330,61 @@ class TwilioConnectionTransport extends StateMachine {
     }
     return false;
   }
+
+  /**
+   * @returns {void}
+   */
+  _setSession(session, sessionTimeout) {
+    this._session = session;
+    this._sessionTimeoutMS = sessionTimeout * 1000;
+  }
+
+  /**
+   * Determines if we should attempt reconnect.
+   * returns a Promise to wait on before attempting to
+   * reconnect. returns null if its not okay to reconnect.
+   * @private
+   * @returns {Promise<void>}
+   */
+  _getReconnectTimer() {
+    if (this._sessionTimeoutMS === 0) {
+      // this means either we have never connected.
+      // or we timed out while trying to reconnect
+      // In either case we do not want to reconnect.
+      return null;
+    }
+
+    // start session timer
+    if (!this._sessionTimer) {
+      this._sessionTimer = setTimeout(() => {
+        // ensure that _clearReconnectTimer wasn't
+        // called while we were waiting.
+        if (this._sessionTimer) {
+          // do not allow any more reconnect attempts.
+          this._sessionTimeoutMS = 0;
+        }
+      }, this._sessionTimeoutMS);
+    }
+
+    // return promise that waits with exponential backoff.
+    this._reconnectAttempt++;
```

Comment:
I think the backoff logic should be the same as the ICE restart backoff logic in PeerConnectionV2.

Reply:
Thanks, I have updated to use `Backoff` module. Can you take another look ?